Objects/abstract.c

```diff
@@ -819,6 +819,20 @@ binary_op(PyObject *v, PyObject *w, const int op_slot, const char *op_name)
     PyObject *result = binary_op1(v, w, op_slot);
     if (result == Py_NotImplemented) {
         Py_DECREF(result);
+
+        if (op_slot == NB_SLOT(nb_rshift) && \
+                strcmp(v->ob_type->tp_name, "builtin_function_or_method") == 0) {
```

Comment:
Yes, but since this code would be slow and complex, I think it can be replaced by something like
```
PyCFunction_Check(v) && PyCFunction_GET_FUNCTION(v) == builtin_print
```
or
```
PyCFunction_Check(v) && strcmp(((PyCFunctionObject *)v) -> m_ml -> ml_name, "print") == 0
```